MetaMask/metamask-extension
# [TS dashboard] Syntax errors don't kill watcher

If you run `yarn ts-migration:dashboard:watch` to compile the TS
migration dashboard app as files being updated, but then you update a
file to cause a syntax error, the watcher will die. This commit prevents
that from happening.

## Manual Testing Steps

* Run `yarn ts-migration:dashboard:watch` in one terminal tab.
* Update `development/ts-migration-dashboard/app/components/App.tsx` to introduce a syntax error. The watcher command should not end.
* Fix this error. You should see "Detected change, rebuilding..." followed by "Compiled \<file type\>:" messages.
* Update `development/ts-migration-dashboard/app/styles/custom-elements.scss` to introduce a syntax error. The watcher command should not end.
* Fix this error. You should see "Detected change, rebuilding..." followed by "Compiled \<file type\>:" messages.
* Introduce a syntax error in either file above again.
* Run `yarn ts-migration:dashboard:build`. You should see an error displayed.
* Fix the syntax error. Run `yarn ts-migration:dashboard:build` again. You should see "Compiled \<file type\>:" messages as above.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
